src/main/java/htsjdk/samtools/DuplicateScoringStrategy.java

```diff
@@ -80,6 +81,8 @@ public static short computeDuplicateScore(final SAMRecord record, final ScoringS
                         score += SAMUtils.getMateCigar(record).getReferenceLength();
                     }
                     break;
+                case RANDOM:
+                    score += (short) Math.sqrt(record.getReadName().hashCode());
```

Comment:
Since the seed is final I think you can do it however you like, in the
enum, or in DuplicateScoringStrategy. Not sure why it needs to go in the
enum.

On Fri, Aug 19, 2016 at 2:02 PM, Tim Fennell <notifications@github.com>
wrote:

> In src/main/java/htsjdk/samtools/DuplicateScoringStrategy.java
> <https://github.com/samtools/htsjdk/pull/688#discussion_r75525818>:
>
> > @@ -80,6 +81,8 @@ public static short computeDuplicateScore(final SAMRecord record, final ScoringS
> >                          score += SAMUtils.getMateCigar(record).getReferenceLength();
> >                      }
> >                      break;
> > +                case RANDOM:
> > +                    score += (short) Math.sqrt(record.getReadName().hashCode());
>
> I can make that change if you'd like. I felt a bit funny about it because
> the enum would have to have an instance of Murmur3, which felt funny to
> me. But other than for the seed, Murmur3 is stateless so it would work.
> Thoughts?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/samtools/htsjdk/pull/688/files/0c6a04385f080299424e60cd54ca4bd8d1c20115#r75525818>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACnk0rUYgxokAty3327sU7gpeDiSL6Cnks5qhe-6gaJpZM4Joo0U>
> .
>